src/java/org/apache/cassandra/cql3/statements/SelectStatement.java

```diff
@@ -247,7 +248,8 @@ public void validate(ClientState state) throws InvalidRequestException
         int pageSize = options.getPageSize();
 
         Selectors selectors = selection.newSelectors(options);
-        ReadQuery query = getQuery(options, selectors.getColumnFilter(), nowInSec, userLimit, userPerPartitionLimit, pageSize);
+        ReadQuery query = getQuery(options, ClientState.forInternalCalls(), selectors.getColumnFilter(),
```

Comment:
Here we should use the `ClientState` that is contained in the `QueryState` argument, so it would be:
```suggestion
        ReadQuery query = getQuery(options, state.getClientState(), selectors.getColumnFilter(),
```

Reply:
👍